examples/online-shop/service.bal

```diff
@@ -0,0 +1,154 @@
+
+final oracledb:Client dbClient = check new(host = HOST, user = USER, password = PASSWORD, port = PORT,
+    database = DATABASE, connectionPool = {maxOpenConnections: 3, minIdleConnections: 1});
+listener http:Listener onlineShopListener = new (9090);
+
+service /onlineshop on onlineShopListener {
+    resource function get employees() returns Employee[]|error? {
+        Employee[] employees = [];
+         stream<Employee, error?> resultStream = dbClient->query(`SELECT * FROM EMPLOYEES`);
+         _ = check resultStream.forEach(function(Employee employee) {
+            employees.push(employee);
+         });
+         check resultStream.close();
+         return employees;
+    }
+
+    resource function get employees/[int id]() returns Employee|error? {
+        return check dbClient->queryRow(`SELECT * FROM EMPLOYEES WHERE EMPLOYEE_ID = ${id}`, Employee);
+    }
+
+    resource function get employees/count() returns int|error? {
+        return check dbClient->queryRow(`SELECT COUNT(*) FROM EMPLOYEES`, int);
+    }
+
+    resource function get products() returns Product[]|error? {
+        Product[] products = [];
+        stream<Product, error?> resultStream = dbClient->query(`SELECT * FROM PRODUCTS`);
+        _ = check resultStream.forEach(function(Product product) {
+           products.push(product);
+        });
+        check resultStream.close();
+        return products;
+    }
+
+    resource function get products/[int id]() returns Product|error? {
+        return check dbClient->queryRow(`SELECT * FROM PRODUCTS WHERE PRODUCT_ID = ${id}`, Product);
+    }
+
+    resource function get customers() returns Customer[]|error? {
+        Customer[] customers = [];
+        stream<Customer, error?> resultStream = dbClient->query(`SELECT * FROM CUSTOMERS`);
+        _ = check resultStream.forEach(function(Customer customer) {
+           customers.push(customer);
+        });
+        check resultStream.close();
+        return customers;
+    }
+
+    resource function get customers/[int id]() returns Customer|error? {
+        return check dbClient->queryRow(`SELECT * FROM CUSTOMERS WHERE CUSTOMER_ID = ${id}`, Customer);
+    }
+
+    resource function post customers(@http:Payload CustomerInput customer) returns string|error? {
+        sql:ParameterizedQuery query = `INSERT INTO CUSTOMERS (NAME,ADDRESS,CREDIT_LIMIT,WEBSITE) VALUES
+        (${customer.name}, ${customer.address}, ${customer.credit_limit}, ${customer.website})`;
+        sql:ExecutionResult result = check dbClient->execute(query);
+        return result.lastInsertId.toString();
+    }
+
+    resource function delete customers/[int id]() returns string|error? {
+        sql:ExecutionResult result = check dbClient->execute(`DELETE FROM CUSTOMERS WHERE CUSTOMER_ID = ${id}`);
+        return result.affectedRowCount.toString();
+    }
+
+    resource function update customers(@http:Payload Customer customer) returns string|error? {
```

Comment:
There is no update method in HTTP, `put` is the correct method for update resources. we need to pass the id as a path param.

Reply:
use put method in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/36/commits/669df6defa5946c90f034656ef97f68df60f7d4e